packages/client/src/store/resolvers.ts

```diff
@@ -74,3 +74,56 @@ export function getAbility( name: string ) {
 		}
 	};
 }
+
+/**
+ * Resolver for getAbilityCategories selector.
+ * Fetches all categories from the server.
+ *
+ * The resolver only fetches once and stores all categories.
+ */
+export function getAbilityCategories() {
+	// @ts-expect-error - registry types are not yet available
+	return async ( { dispatch, registry, select } ) => {
+		const existingCategories = select.getAbilityCategories();
```

Comment:
It's probably similar to https://github.com/WordPress/abilities-api/pull/121/files#r2437266219, but here I have some concerns. If someone registers first a single category from the client, then this resolver will never hit the server, as it will find an entry. 

One solution would be to have two selectors, one private (or explicitly tied to the server) for network requests without any conditions letting the framework handle caching. Another selector wouldn't have the matching resolver, but would combine categories from the client and the server together.

Another option, filter out client-only categories in the check.

Reply:
See https://github.com/WordPress/abilities-api/pull/121/commits/91d8bf8fd7131bedb38400a2f2199291c72d6442. I opted for the 3rd option, `filter out client-only categories in the check` which matches how we are handling `getAbilities` as well.